4-HTTP/client/main.js

```diff
@@ -0,0 +1,209 @@
+        createItem(todos.data, newTodo),
+        todoList.firstChild
+      );
+    } catch (error) {
+      console.error(error);
+    }
+    inputText.value = "";
+  };
+
+  const removeTodo = async (item) => {
+    document.getElementById(item.id).remove();
+    try {
+      await todosAPI.removeTodo(item.id);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  function removeElement(array, elem) {
+    const index = array.findIndex((el) => el.id == elem.id);
+    if (index > -1) array.splice(index, 1);
+  }
+
+  const swapDOMComponents = (elem, curr, modifiedElem) => {
+    elem
+      .querySelector(`.${classes.listItemText}`)
+      .replaceChild(curr, modifiedElem);
+  };
+
+  const createTodoItem = (item, todoListItem) => {
+    const todoItem = document.createElement("span");
+    todoItem.innerText = document.getElementById(`${item.id}-editTodo`).value;
+    todoItem.classList.add(classes.todoText);
+    if (item.isFinished) {
+      todoItem.classList.add(classes.finishedTodo);
+      todoListItem.firstChild.checked = true;
+    }
+    return todoItem;
+  };
+
+  const createEditTodoElem = (elemToChange, item) => {
+    const inputText = document.createElement("input");
+    inputText.type = "text";
+    inputText.value = elemToChange.innerText;
+    inputText.classList.add(classes.todoText, classes.editTodo);
+    inputText.id = `${item.id}-editTodo`;
+    return inputText;
+  };
+
+  const editTodo = async (item) => {
+    const todoListItem = document.getElementById(item.id);
+    const isFound = editModeItems.some((element) => element.id === item.id);
+    const elemToChange = todoListItem.querySelector(`.${classes.todoText}`);
+    if (isFound) {
+      removeElement(editModeItems, item);
+      const todoItem = createTodoItem(item, todoListItem);
+      swapDOMComponents(todoListItem, todoItem, elemToChange);
+      try {
+        await todosAPI.editTodo(
+          { ...item, text: todoItem.innerText, isFinished: item.isFinished },
+          item.id
+        );
+      } catch (error) {
+        console.log(error);
+      }
+    } else {
+      editModeItems.push(item);
+      const inputText = createEditTodoElem(elemToChange, item);
+      swapDOMComponents(todoListItem, inputText, elemToChange);
+    }
+  };
+
+  const toggleTodo = async (item) => {
+    const todoListItem = document.getElementById(item.id);
+    const todoDescription = todoListItem.firstChild.children[1];
+    if (item.isFinished) {
+      todoDescription.classList.remove(classes.finishedTodo);
+      todoDescription.classList.add(classes.unfinishedTodo);
+      item.isFinished = false;
+    } else {
+      todoDescription.classList.remove(classes.unfinishedTodo);
+      todoDescription.classList.add(classes.finishedTodo);
+      item.isFinished = true;
+    }
+    try {
+      await todosAPI.editTodo(item, item.id);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const createButton = (cb, icon, className) => {
```

Comment:
If it's an Icon button
```suggestion
  const createIconButton = (cb, icon, className) => {
```

Reply:
Thanks for the suggestion, I will fix that.